src/day8/re_frame_10x/components/cljs_devtools.cljs

```diff
@@ -334,4 +400,124 @@
+          copy-repl-item   (create-menu-item "Copy REPL command")
+          element-rect     (.getBoundingClientRect html-element)
+          x-pos            (+ (.-left element-rect) (.-scrollX js/window))
+          y-pos            (+ (.-top element-rect) (.-scrollY js/window))]
+      (doto copy-path-item
+        (.addClassName "copy-path")
+        (.addClassName "10x-menu-item"))
+      (doto copy-obj-item
+        (.addClassName "copy-object")
+        (.addClassName "10x-menu-item"))
+      (doto copy-repl-item
+        (.addClassName "copy-repl")
+        (.addClassName "10x-menu-item"))
+      (doto popup-menu
+        (.addItem copy-path-item)
+        (.addItem copy-obj-item)
+        (.addItem copy-repl-item)
+        (.showAt x-pos y-pos)
+        (.render html-element))
+      (goog.object.forEach
+        goog.ui.Component.EventType
+        (fn [type]
+          (goog.events.listen
+            popup-menu
+            type
+            (fn [e] (cond
+                      (= (.-type e) "hide")
+                      (when-not (or (empty? @event-log) (= (peek @event-log) "unhighlight"))
+                        ;; if the last event registered is 'unhighlight' then we should close the dialog
+                        ;; `unhighlight` is dispatched when the mouse leaves a menu item
+                        ;; if the mouse icon would scroll over another item in the menu, a `highlight` event would
+                        ;; overwrite the last `unhighlight`
+                        (.preventDefault e))
+
+                      (= (.-type e) "action")
+                      (let [class-names (-> e .-target .getExtraClassNames js->clj)]
+                        (swap! event-log conj "action")
+                        (cond
+                          (some (fn [class-name] (= class-name "copy-object")) class-names)
+                          (let [path-obj (reader.edn/read-string-maybe path)
+                                nested?  (when viewing-path
+                                           (count viewing-path))
+                                path-obj (if nested?
+                                           (subvec path-obj nested?) ;; both path and viewing-path are from the root point of view
+                                           path-obj)
+                                object   (get-in data path-obj)]
+                            ;; note we cant copy nil objects
+                            (if (or object (and (not (nil? object)) (= object false)))
+                              (do (clipboard/copy! object)
+                                  (js/console.log "Copied object"))
+                              (js/console.error "Could not copy!")))
+
+                          (some (fn [class-name] (= class-name "copy-path")) class-names)
+                          (do
+                            (clipboard/copy! path)
+                            (js/console.log "Copied path"))
```

Comment:
should we remove these console.log ?

Reply:
Sure we can, I added them to signal change just incase a user wanted to confirm.